src/uct/cuda/cuda_copy/cuda_copy_iface.c

```diff
@@ -32,6 +33,14 @@ static ucs_config_field_t uct_cuda_copy_iface_config_table[] = {
      "Max number of cuda events. -1 is infinite",
      ucs_offsetof(uct_cuda_copy_iface_config_t, max_cuda_events), UCS_CONFIG_TYPE_UINT},
 
+    {"MEMTYPE_CACHE", "y",
+     "Enable memory type cache in cuda_copy UCT\n",
+     ucs_offsetof(uct_cuda_copy_iface_config_t, enable_memtype_cache), UCS_CONFIG_TYPE_BOOL},
+
+    {"DETECT_THRESH", "128k",
+     "Threshold for detecting src/dst mem_types for stream picking",
```

Comment:
Threshold for detecting buffer memory types for picking the best stream to use. Messages smaller than this value will pick a random stream.